### PR TITLE
correctly export wallet data

### LINKED ISSUE #2024: [bug] wallet export / import use the different format
when I export my wallet addr:
```sh
go-filecoin wallet export fcqhn.....
```
got
```
Address:	fcqhn.....
PrivateKey:	ceaf5e302....
Curve:		secp256k1
```

or use `--enc=json` get a json output, but
```sh
go-filecoin wallet import <walletFile> 
```
got
```
Error: invalid length, need 256 bits
```

why not wallet export / import use the same format

### PR DESCRIPTION
**Pull request by frrist**
### Whats this?

This PR aligns the type `go-filecoin wallet export --enc=json` returns and the type `go-filecoin wallet import` expects. 

#2024 was caused by the `export` command returning:
`{"privateKey":"xVHXgLfpcJ6kisQONRelKHA5DuY+cpIvTTK9K3utxDI=","curve":"secp256k1"}`

And the `import` command expexting:
`{"KeyInfo":[{"privateKey":"xVHXgLfpcJ6kisQONRelKHA5DuY+cpIvTTK9K3utxDI=","curve":"secp256k1"}]}`

closes #2024

### CODE REVIEW

File: api/address.go [hunk 1]
```diff
@@ -13,7 +13,8 @@ import (
 // Address is the interface that defines methods to manage Filecoin addresses and wallets.
 type Address interface {
```
Comment (anorth) [inline api/address.go]: Aside: this is a confusing name for an interface that contains multiple addresses.

File: api/impl/address.go [hunk 2]
```diff
@@ -57,14 +57,19 @@ func (api *nodeAddrs) Lookup(ctx context.Context, addr address.Address) (peer.ID
...
 
-	kinfos, err := parseKeyInfos(d)
+	kinfos, err := parseKeyInfos(f)
 	if err != nil {
 		return nil, err
 	}
 
+	// error if we fail to parse keyinfo from the provided file.
+	if len(kinfos) == 0 {
+		return nil, fmt.Errorf("failed to parse keys from wallet file")
```
Comment (anorth) [inline api/impl/address.go]: Be more specific "no keys in wallet file %s"

File: commands/address.go [hunk 3]
```diff
@@ -142,7 +143,17 @@ var walletImportCmd = &cmds.Command{
...
 	Run: func(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment) error {
-		addrs, err := GetAPI(env).Address().Import(req.Context, req.Files)
+		iter := req.Files.Entries()
+		if !iter.Next() {
+			return fmt.Errorf("no file given: %s", iter.Err())
+		}
+
+		fi, ok := iter.Node().(files.File)
+		if !ok {
+			return fmt.Errorf("given file was not a files.File")
```
Comment (anorth) [inline commands/address.go]: Since this message will go to the user, perhaps avoid Go terms and say "%s is not a file"

File: api/impl/address.go [hunk 4]
```diff
@@ -110,21 +115,15 @@ func (api *nodeAddress) Export(ctx context.Context, addrs []address.Address) ([]
...
-		fi, ok := iter.Node().(files.File)
-		if !ok {
-			return nil, fmt.Errorf("file passed to import was not a file")
-		}
-
-		var ki types.KeyInfo
-		if err := json.NewDecoder(fi).Decode(&ki); err != nil {
-			return nil, err
-		}
+// WalletImportResult is the structure used to hold imported wallet data.
```
Comment (anorth) [inline api/impl/address.go]: Do I understand correctly that this type defines the schema expected for imported wallets? If so make that very explicit, and probably raise this higher up in the file, above the funcs.

File: api/address.go (hunk 1 above)
Comment (frrist) [reply api/address.go]: Agreed, this bit is even more confusing: https://github.com/filecoin-project/go-filecoin/issues/1693.

File: api/impl/address.go (hunk 2 above)
Comment (frrist) [reply api/impl/address.go]: I will state that there are "no keys in the file", but I don't believe the interface for `go-ipfs-files` we are using has a `filename`/`string` methods, meaning I can't return the name of the file :(

Comment (frrist) [reply api/impl/address.go]: Aforementioned files interface: https://ipfs.io/ipfs/QmQmhotPUzVrMEWNK3x1R5jQ5ZHWyL7tVUrmRPjrBrvyCb/go-ipfs-files/file.go

File: commands/address.go (hunk 3 above)
Comment (frrist) [reply commands/address.go]: Same issue as stated here: https://github.com/filecoin-project/go-filecoin/pull/2100#discussion_r260047378

File: api/impl/address.go (hunk 4 above)
Comment (frrist) [reply api/impl/address.go]: The `WalletImportResult` type and [this type](https://github.com/filecoin-project/go-filecoin/blob/master/commands/address.go#L171) are the same -- I think there should be 1 type both commands use. The type cannot live in `commands/` as that will cause an import cycle with `api/impl`. I suppose I can move it to `api/impl` and import that into `commands/`

File: commands/address_daemon_test.go [hunk 5]
```diff
@@ -115,3 +117,26 @@ func TestWalletLoadFromFile(t *testing.T) {
...
 }
+
+func TestWalletExportImportRoundTrip(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	d := th.NewDaemon(t).Start()
+	defer d.ShutdownSuccess()
+
+	dw := d.RunSuccess("address", "ls").ReadStdoutTrimNewlines()
```
Comment (phritz) [inline commands/address_daemon_test.go]: should probably create a few addresses before calling ls. my guess is there might be ordering issues, but maybe not.

Comment (frrist) [reply commands/address_daemon_test.go]: The ordering on import and export shouldn't matter, the problem was the way the exported structure was serialized to json didn't match what import expected.

**Pull request merged**